src/java/org/apache/cassandra/audit/AuditLogManager.java

```diff
@@ -80,14 +85,14 @@ public void initialize()
             registerAsListener();
     }
 
-    private IAuditLogger getAuditLogger(String loggerClassName) throws ConfigurationException
+    private IAuditLogger getAuditLogger(ParameterizedClass logger) throws ConfigurationException
     {
-        if (loggerClassName != null)
+        if (logger.class_name != null)
         {
-            return FBUtilities.newAuditLogger(loggerClassName);
+            return FBUtilities.newAuditLogger(logger.class_name, logger.parameters);
         }
 
-        return FBUtilities.newAuditLogger(BinAuditLogger.class.getName());
+        return FBUtilities.newAuditLogger(BinAuditLogger.class.getName(), null);
```

Comment:
nit: Did you mean to use `null` here as this seems to be inconsistent with usage of `Collections.emptyMap()` in other places in this PR for class parameters?

Reply:
fixed, thanks, I covered "custom logger" case as well when map is null, so the contract here is that Map will never be null and it will be at least empty.